warp10/src/main/java/io/warp10/script/ext/http/HTTP.java

```diff
@@ -0,0 +1,449 @@
+
+    // If not defined, use already existing StandaloneWebCallService webAccessController which uses Configuration.WEBCALL_HOST_PATTERNS
+    if (null == patternConf) {
+      webAccessController = StandaloneWebCallService.getWebAccessController();
+    } else {
+      webAccessController = new WebAccessController(patternConf);
+    }
+
+    // retrieve authentication required
+    auth = "true".equals(WarpConfig.getProperty(HttpWarpScriptExtension.WARPSCRIPT_HTTP_AUTHENTICATION_REQUIRED));
+
+    // retrieve capName
+    capName = WarpConfig.getProperty(HttpWarpScriptExtension.WARPSCRIPT_HTTP_CAPABILITY);
+
+    // retrieve limits
+    Object confMaxRequests = WarpConfig.getProperty(HttpWarpScriptExtension.WARPSCRIPT_HTTP_REQUESTS);
+    if (null == confMaxRequests) {
+      baseMaxRequests = HttpWarpScriptExtension.DEFAULT_HTTP_REQUESTS;
+    } else {
+      baseMaxRequests = Long.parseLong((String) confMaxRequests);
+    }
+
+    Object confMaxSize = WarpConfig.getProperty(HttpWarpScriptExtension.WARPSCRIPT_HTTP_SIZE);
+    if (null == confMaxSize) {
+      baseMaxSize = HttpWarpScriptExtension.DEFAULT_HTTP_MAXSIZE;
+    } else {
+      baseMaxSize = Long.parseLong((String) confMaxSize);
+    }
+  }
+
+  @Override
+  public Object apply(WarpScriptStack stack) throws WarpScriptException {
+
+    Object o = stack.pop();
+    if (!(o instanceof Map)) {
+      throw new WarpScriptException(getName() + " expects a MAP as input.");
+    }
+
+    Map params = (Map) o;
+
+    //
+    // Check authorization
+    //
+
+    if (auth && !stack.isAuthenticated()) {
+      throw new WarpScriptException(getName() + " requires the stack to be authenticated.");
+    }
+
+    if (null != capName && null == Capabilities.get(stack, capName)) {
+      throw new WarpScriptException("Capability " + capName + " is required by function " + getName());
+    }
+
+    //
+    // Retrieve call number limit and download size limit
+    //
+
+    long maxrequests;
+    if (null != Capabilities.get(stack, HttpWarpScriptExtension.ATTRIBUTE_HTTP_REQUESTS)) {
+      maxrequests = Long.valueOf(Capabilities.get(stack, HttpWarpScriptExtension.ATTRIBUTE_HTTP_REQUESTS));
+    } else {
+      maxrequests = baseMaxRequests;
+    }
+
+    long maxsize;
+    if (null != Capabilities.get(stack, HttpWarpScriptExtension.ATTRIBUTE_HTTP_SIZE)) {
+      maxsize = Long.valueOf(Capabilities.get(stack, HttpWarpScriptExtension.ATTRIBUTE_HTTP_SIZE));
+    } else {
+      maxsize = baseMaxSize;
+    }
+
+    //
+    // Retrieve arguments
+    //
+
+    String method = (String) params.get(METHOD);
+    if (null == method) {
+      throw new WarpScriptException(getName() + " expects an http method.");
+    }
+
+    Map<Object, Object> headers = (Map) params.getOrDefault(HEADERS, new HashMap<>());
+    Object body = params.get(BODY);
+
+    List authInfo = (List) params.get(AUTH_INFO);
+    WarpScriptStack.Macro authMacro = (WarpScriptStack.Macro) params.get(AUTH_MACRO);
```

Comment:
You are not checking if the parameter is a Macro, this will throw an exception if that is not the case